Source/AwsCommonRuntimeKit/io/Stream.swift

```diff
@@ -4,106 +4,99 @@
 import AwsCIo
 import Foundation
 
-// swiftlint:disable trailing_whitespace
+/// This is for streaming input.
+public protocol IStreamable {
 
-private var vtable = aws_input_stream_vtable(seek: doSeek,
-        read: doRead,
-        get_status: doGetStatus,
-        get_length: doGetLength,
-        acquire: { _ = Unmanaged<AwsInputStream>.fromOpaque($0!.pointee.impl).retain() },
-        release: { Unmanaged<AwsInputStream>.fromOpaque($0!.pointee.impl).release() })
+    /// Optional, throws length not supported error by default
+    func length() throws -> UInt64
```

Comment:
Why provide a default implementation if it just throws an error by default?

Reply:
This code is called by C. It can have some alternate implementation if some stream doesn't support length.